posthog/cloud_utils.py

```diff
@@ -38,3 +44,45 @@ def is_cloud():
 def TEST_clear_cloud_cache(value: Optional[bool] = None):
     global is_cloud_cached
     is_cloud_cached = value
+
+
+def get_cached_instance_license() -> Optional["License"]:
+    """Returns the first valid license and caches the value for the lifetime of the instance, as it is not expected to change.
+    If there is no valid license, it returns None.
+    """
+    global instance_license_cached
+    global is_instance_licensed_cached
+
+    try:
+        from ee.models.license import License
+
+        if isinstance(instance_license_cached, License):
+            return instance_license_cached
+
+        if is_instance_licensed_cached is False:
+            # This is an unlicensed instance
+            return None
+
+        # TRICKY - The license table may not exist if a migration is running
+        license = License.objects.first_valid()
+        if license:
+            instance_license_cached = license
+            is_instance_licensed_cached = True
+        return instance_license_cached
+
+    except ProgrammingError:
+        # TRICKY - The license table may not exist if a migration is running
+        pass
+    except Exception as e:
+        print("ERROR: Unable to check license", e)  # noqa: T201
```

Comment:
This function seems simple enough that I'm worried this is too lax. It already hid the problem with undefined `instance_license_cached`, so I think it's better to raise errors. The `print` is also unlikely to be useful at all here – better to have a problem in Sentry, if there ever is one